### PR TITLE
Refactor ConfigCache and related pathways ,uniformly use UTF-8 encoding for md5 fields

### PR DESCRIPTION
**Pull request by Sunrisea**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Refactor ConfigCache and related pathways ,uniformly use UTF-8 encoding for md5 fields. Utilize the SPI mechanism in related pathways and methods to ensure extensibility for support of other types of encoding. Fix test.

## Brief changelog

Create the ConfigCacheFactory interface to handle the creation of ConfigCache instances. 
Create the Md5PostProcessor interface to handle the md5 dump of ConfigCache.
Create the Md5Comparator interface.
Utilize the SPI (Service Provider Interface) mechanism in related pathways and methods to ensure extensibility for support of other types of encoding, and implement the default handling methods for UTF-8 encoded md5 values.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [*] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [*] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [*] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [*] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [*] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by wuyfee** [comment]
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11948842979)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ❌ - e2e-cpp-test (standalone & cluster):   failure 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactory.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    public ConfigCache createConfigCache();
+    
+    /**
+     * Create config cache config cache.
+     *
+     * @param md5            the md 5
+     * @param lastModifiedTs the last modified ts
+     * @return the config cache
+     */
+    public ConfigCache createConfigCache(String md5, long lastModifiedTs);
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactory.java:40]: interface中减少方法数量，仅透出createConfigCache()，createConfigCacheGray() ,
比如

可在次基础上在ConfigCacheFactoryDelegate中重载提供四个方法，内部调用ConfigCacheFactory中的原子方法进行二次封装

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public ConfigCache createConfigCache() {
+        return configCacheFactory.createConfigCache();
+    }
+    
+    public ConfigCache createConfigCache(String md5, long lastModifiedTs) {
+        return configCacheFactory.createConfigCache(md5, lastModifiedTs);
+    }
+    
+    public ConfigCacheGray createConfigCacheGray(String grayName) {
+        return configCacheFactory.createConfigCacheGray(grayName);
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java:75]: 改为
ConfigCacheGray configGray=configCacheFactory.createConfigCacheGray()；
configGray.setGrayName(grayName);

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    public static ConfigCacheFactoryDelegate getInstance() {
+        return INSTANCE;
+    }
+    
+    public ConfigCache createConfigCache() {
+        return configCacheFactory.createConfigCache();
+    }
+    
+    public ConfigCache createConfigCache(String md5, long lastModifiedTs) {
+        return configCacheFactory.createConfigCache(md5, lastModifiedTs);
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java:71]: 改为
ConfigCache configGray=this.createConfigCachhe()；
configGray.setMd5(md5);
configGray.setlastModifiedTs(lastModifiedTs);

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheMd5PostProcessor.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package com.alibaba.nacos.config.server.model;
+
+/**
+ * The interface Config cache md5 post processor.
+ *
+ * @author Sunrisea
+ */
+public interface ConfigCacheMd5PostProcessor {
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheMd5PostProcessor.java:24]: 类名上可以不要透出md5

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11949536662)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11949763379)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11951540635)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11952476247)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11952961208)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            if (StringUtils.isEmpty(each.getConfigCacheFactoryName())) {
+                LOGGER.warn("[ConfigCacheFactory] Load ConfigCacheFactory({}) ConfigFactroyName (null/empty) fail. "
+                                + "Please add ConfigFactoryName to resolve",
+                        each.getClass());
+                continue;
+            }
+            LOGGER.info("[ConfigCacheFactory] Load ConfigCacheFactory({}) ConfigCacheFactoryName({}) successfully. ",
+                    each.getClass(), each.getConfigCacheFactoryName());
+            if (StringUtils.equals(configCacheFactoryType, each.getConfigCacheFactoryName())) {
+                this.configCacheFactory = each;
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactoryDelegate.java:54]: add log , load specific config cache factory successfully with name {name}

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheGray.java [hunk 6]
```diff
@@ -49,15 +51,6 @@ public GrayRule getGrayRule() {
...
-    public ConfigCacheGray(String md5Gbk, String md5Utf8, long lastModifiedTs, String grayRule)
-            throws RuntimeException {
-        super(md5Gbk, md5Utf8, lastModifiedTs);
-        this.grayRule = GrayRuleManager.constructGrayRule(GrayRuleManager.deserializeConfigGrayPersistInfo(grayRule));
-        if (this.grayRule == null || !this.grayRule.isValid()) {
-            throw new RuntimeException("raw gray rule is invalid");
-        }
-    }
-    
     public String getGrayName() {
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheGray.java:54]: miss code：

this.grayRule = GrayRuleManager.constructGrayRule(GrayRuleManager.deserializeConfigGrayPersistInfo(grayRule));
        if (this.grayRule == null || !this.grayRule.isValid()) {
            throw new RuntimeException("raw gray rule is invalid");
}

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/12006924432)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Pull request closed without merging**